experiment/commenter/main.go

```diff
@@ -0,0 +1,167 @@
+	flag.BoolVar(&o.wet, "wet", false, "Mutate github if set")
+	flag.StringVar(&o.comment, "comment", "", "Append the following comment to matching issues")
+	flag.IntVar(&o.ceiling, "ceiling", 3, "Maximum number of issues to modify, 0 for infinite")
+	flag.StringVar(&o.user, "user", "", "Github username of commenter")
+	flag.StringVar(&o.token, "token", "", "Path to github token")
+	flag.Parse()
+	return o
+}
+
+func parseHTMLURL(url string) (string, string, int, error) {
+	// https://github.com/batterseapower/pinyin-toolkit/issues/132`)
+	re := regexp.MustCompile(`.+/(.+)/(.+)/(issues|pull)/(\d+)$`)
+	mat := re.FindStringSubmatch(url)
+	if mat == nil {
+		return "", "", 0, fmt.Errorf("failed to parse: %s", url)
+	}
+	n, err := strconv.Atoi(mat[4])
+	if err != nil {
+		return "", "", 0, err
+	}
+	return mat[1], mat[2], n, nil
+}
+
+func makeQuery(query string, includeClosed bool, minUpdated time.Duration) string {
+	parts := []string{query}
+	if !includeClosed {
+		parts = append(parts, "is:open")
+	}
+	if minUpdated != 0 {
+		latest := time.Now().Add(-minUpdated)
+		parts = append(parts, "updated:<="+latest.Format(time.RFC3339))
+	}
+	return strings.Join(parts, " ")
+}
+
+type options struct {
+	asc           bool
+	ceiling       int
+	comment       string
+	includeClosed bool
+	query         string
+	sort          string
+	token         string
+	updated       time.Duration
+	user          string
+	wet           bool
+}
+
+type client interface {
+	CreateComment(owner, repo string, number int, comment string) error
+	FindIssues(query, sort string, asc bool) ([]github.Issue, error)
+}
+
+func main() {
+	log.SetFlags(log.LstdFlags | log.Lshortfile)
+	o := flagOptions()
+
+	if o.query == "" {
+		log.Fatal("empty --query")
+	}
+	if o.user == "" {
+		log.Fatal("empty --user")
+	}
+	if o.token == "" {
+		log.Fatal("empty --token")
+	}
+	b, err := ioutil.ReadFile(o.token)
+	if err != nil {
+		log.Fatalf("cannot read --token: %v", err)
+	}
+
+	var c client
+	tok := strings.TrimSpace(string(b))
+	if o.wet {
+		c = github.NewClient(o.user, tok)
+	} else {
+		c = github.NewDryRunClient(o.user, tok)
+	}
+
+	query := makeQuery(o.query, o.includeClosed, o.updated)
+	sort := ""
+	asc := false
+	if o.updated > 0 {
+		sort = "updated"
+		asc = true
+	}
+	if err := run(c, query, sort, asc, o.comment, o.ceiling); err != nil {
+		log.Fatalf("Failed run: %v", err)
+	}
+}
+
+func run(c client, query, sort string, asc bool, comment string, ceiling int) error {
+	log.Printf("Searching: %s", query)
+	issues, err := c.FindIssues(query, sort, asc)
+	if err != nil {
+		return fmt.Errorf("search failed: %v", err)
+	}
+	problems := []string{}
+	log.Printf("Found %d matches", len(issues))
+	for n, i := range issues {
+		if ceiling > 0 && n == ceiling {
+			log.Printf("Stopping at --ceiling=%d of %d results", n, len(issues))
+			break
+		}
+		log.Printf("Matched %s: %s", i.HTMLURL, i.Title)
+		org, repo, number, err := parseHTMLURL(i.HTMLURL)
```

Comment:
Why not just grab this info from `i`? It seems unnecessary to go through the HTMLURL.